simulation/agents.py

```diff
@@ -475,10 +483,18 @@ def install_heating_system(
         self.heating_system = heating_system
         self.heating_system_install_date = model.current_datetime.date()
 
+        if self.boiler_upgrade_grant_available:
+            if heating_system == HeatingSystem.HEAT_PUMP_AIR_SOURCE:
+                self.boiler_upgrade_grant_used = 5_000
+            if heating_system == HeatingSystem.HEAT_PUMP_GROUND_SOURCE:
+                self.boiler_upgrade_grant_used = 6_000
+
     def update_heating_status(self, model: "DomesticHeatingABM") -> None:
 
         self.epc_c_upgrade_costs = {}
         self.heating_system_total_costs = {}
+        self.boiler_upgrade_grant_available = False
+        self.boiler_upgrade_grant_used = 0
```

Comment:
That's right, it's just to identify the relevant timestep. Else we have a bunch of subsidy values populated for a given agent for every time-step after which they used a grant in the output file, which is a bit strange. `model.total_subsidies_spent` won't be a sum of zeroes because in `household.make_decisions()`, the heating system choice follows the household's update of their heating status -- i.e. this reset happens on the step following the heating decision. We used the same pattern for `household.is_heating_functioning`

Reply:
Ah yeah ok, makes sense. I got confused as in `model.total_subsidies_spent` you are doing `+=` (which is correct) whereas I thought you were doing just `=` and so resetting it wasn't the right thing to do.

But I agree this approach is correct!